src/app/clients/add/formComponents.tsx

```diff
@@ -0,0 +1,534 @@
+    addressLine2: "",
+    addressTown: "",
+    addressCounty: "",
+    addressPostcode: "",
+    adults: [
+        { personType: "adult", quantity: 0 },
+        { personType: "male", quantity: 0 },
+        { personType: "female", quantity: 0 },
+    ],
+    numberChildren: 0,
+    children: [],
+    dietaryRequirements: {},
+    feminineProducts: {},
+    babyProducts: null,
+    nappySize: "",
+    petFood: {},
+    otherItems: {},
+    deliveryInstructions: "",
+    extraInformation: "",
+};
+
+const initialFormErrors: FormErrors = {
+    fullName: Error.initial,
+    phoneNumber: Error.none,
+    addressLine1: Error.initial,
+    addressPostcode: Error.initial,
+    adults: Error.initial,
+    numberChildren: Error.initial,
+    nappySize: Error.none,
+};
+
+const AddClientForm: React.FC = () => {
+    const router = useRouter();
+    const [fields, setFields] = useState<Fields>(initialFields);
+    const [formErrors, setFormErrors] = useState<FormErrors>(initialFormErrors);
+    const [submitError, setSubmitError] = useState<Error>(Error.none);
+
+    useEffect(() => {
+        const childrenCopy: Person[] = [];
+        for (let index = 0; index < fields.numberChildren; index++) {
+            if (index < fields.children.length) {
+                childrenCopy.push(fields.children[index]);
+            } else {
+                childrenCopy.push({ personType: "child", age: null, quantity: 1 });
+            }
+        }
+        fieldSetter("children", childrenCopy);
+    }, [fields.numberChildren]); // eslint-disable-line
+
+    const fieldSetter = setField(setFields, fields);
+    const errorSetter = setError(setFormErrors, formErrors);
+
+    const submitForm = async (): Promise<void> => {
+        let extraInformationWithNappy = fields.extraInformation;
+        if (fields.nappySize !== "") {
+            extraInformationWithNappy = `Nappy Size: ${fields.nappySize}, Extra Information: ${fields.extraInformation}`;
+        }
+        const clientRecord: ClientDatabaseRecord = {
+            full_name: fields.fullName,
+            phone_number: fields.phoneNumber,
+            address_1: fields.addressLine1,
+            address_2: fields.addressLine2,
+            address_town: fields.addressTown,
+            address_county: fields.addressCounty,
+            address_postcode: fields.addressPostcode,
+            dietary_requirements: checkboxGroupToArray(fields.dietaryRequirements),
+            feminine_products: checkboxGroupToArray(fields.feminineProducts),
+            baby_food: fields.babyProducts,
+            pet_food: checkboxGroupToArray(fields.petFood),
+            other_items: checkboxGroupToArray(fields.otherItems),
+            delivery_instructions: fields.deliveryInstructions,
+            extra_information: extraInformationWithNappy,
+        };
+        console.log(clientRecord);
+        const inputError = checkErrorOnSubmit(formErrors, setFormErrors);
+        let submitErrorValue = Error.none;
```

Comment:
do you need a `let`? can you just `setSubmitError(Error.database)` whenever you need it?